ToNote/Controls/NoteRichTextBoxPanel.cs

```diff
@@ -0,0 +1,182 @@
+            set { SetValue(SaveContentsToFilesCommandProperty, value); }
+        }
+
+        public static readonly DependencyProperty SaveContentsToFilesCommandProperty = DependencyProperty.Register("SaveContentsToFilesCommand",
+           typeof(ICommand), typeof(NoteRichTextBoxPanel), new FrameworkPropertyMetadata(new RelayCommand<NoteRichTextBoxPanel>((panel) =>
+           {
+               var note = panel?.Note;
+
+               if (note == null) return;
+
+               var index = 0;
+
+               var directory = ".\\Notes";
+
+               if (!Directory.Exists(directory))
+                   Directory.CreateDirectory(directory);
+
+               foreach (var child in panel.Items)
+               {
+                   if (!(child is ExtendedRichTextBox rtb)) continue;
+
+                   string file;
+                   bool newfile = false;
+
+                   if (note.FileNames.Count > index && note.FileNames[index] != null)
+                       file = note.FileNames[index];
+                   else
+                   {
+                       file = $"{directory}\\{note.Name.ToLower()}_{index}";
+                       newfile = true;
+                   }
+
+                   using (var stream = new FileStream(file, FileMode.OpenOrCreate))
+                   {
+                       var text = new TextRange(rtb.Document.ContentStart, rtb.Document.ContentEnd);
+                       text.Save(stream, DataFormats.Rtf);
+                   }
+
+                   if (newfile)
+                       note.FileNames.Add(file);
+
+                   index += 1;
+               }
+
+               var serializedNote = JsonConvert.SerializeObject(note);
+               var metadataFileName = $"{note.Name.ToLower()}Metadata.txt";
+               File.WriteAllText(metadataFileName, serializedNote);
+           })));
+
+        /// <summary>
+        /// Applies appropriate events to the provided ExtendedRichTextBox control
+        /// </summary>
+        /// <param name="rtb"></param>
+        private void ConfigureRichTextBoxEvents(ExtendedRichTextBox rtb)
+        {
+            rtb.BackspacePressedWhileEmpty += (s, e) =>
+            {
+                var index = this.Items.IndexOf(Keyboard.FocusedElement);
+
+                SwitchKeyboardFocusToNextRTB(index);
+
+                if (this.Items.Contains(rtb))
+                    this.Items.Remove(rtb);
+
+                Note?.DeleteFile(rtb.CurrentFile);
+            };
+        }
+
+        /// <summary>
+        /// Gives a neighbour ExtendedRichTextBox control the focus.
+        /// </summary>
+        /// <param name="index">Index of current RTB in the Items Array</param>
+        /// <param name="up">Whether to navigate up the list.</param>
+        private void SwitchKeyboardFocusToNextRTB(int index, bool up = true)
+        {
+            if (up)
+            {
+                for (int i = index - 1; i >= 0; i--)
```

Comment:
Same here, could be a bit nicer with linq.
```cs
var rtb = Items.Cast<object>().Take(index).OfType<ExtendedRichTextBox>().LastOrDefault();
if (rtb != null)
    Keyboard.Focus(rtb);
```
(Cast<object>() is needed because ItemCollection only implements non-generic IEnumerable)